e2e/specs/patientSearch.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { test } from '../core';
+import { HomePage } from '../pages/homePage';
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, Patient } from '../commands';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('should be able to search patients by identifier', async ({ page, api }) => {
+  // extract details from the created patient
+  const openmrsIdentifier = patient.identifiers[0].display.split('=')[1].trim();
+  const firstName = patient.person.display.split(' ')[0];
+  const lastName = patient.person.display.split(' ')[1];
+
+  const homePage = new HomePage(page);
+  await homePage.goto();
+
+  await homePage.searchPatient(openmrsIdentifier);
+
+  await expect(homePage.floatingSearchResultsContainer()).toHaveText(/1 search results/);
+  await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(firstName));
+  await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(lastName));
+
+  await homePage.clickOnPatientResult(firstName);
+  await expect(homePage.page).toHaveURL(new RegExp(`/patient/${patient.uuid}/chart`));
+});
+
+test('should be able to search patients by name', async ({ page, api }) => {
+  // extract details from the created patient
+  const openmrsIdentifier = patient.identifiers[0].display.split('=')[1].trim();
+  const firstName = patient.person.display.split(' ')[0];
+  const lastName = patient.person.display.split(' ')[1];
```

Comment:
`patient.person.names[0].givenName` and `patient.person.names[0].familyName` should be preferred.

Reply:
When you make a `POST` request to the `/patient` endpoint, the names are combined to a single string with an attribute called display. That's why I needed to separate them. Here is an example response:

Here's an example payload: 
```json
{
    "uuid": "9b6b0edd-5494-43c5-a69d-ddcc5372385b",
    "display": "1000489 - Westtt Patient",
    "identifiers": [
        {
            "uuid": "dc5a044b-54be-4f92-a9fc-d5707c477af4",
            "display": "OpenMRS ID = 1000489",
            "links": [
                {
                    "rel": "self",
                    "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/patient/9b6b0edd-5494-43c5-a69d-ddcc5372385b/identifier/dc5a044b-54be-4f92-a9fc-d5707c477af4",
                    "resourceAlias": "identifier"
                }
            ]
        }
    ],
    "person": {
        "uuid": "9b6b0edd-5494-43c5-a69d-ddcc5372385b",
        "display": "Westtt Patient",
        "gender": "M",
        "age": 0,
        "birthdate": "2022-06-01T00:00:00.000+0000",
        "birthdateEstimated": false,
        "dead": false,
        "deathDate": null,
        "causeOfDeath": null,
        "preferredName": {
            "uuid": "25acf26b-74c6-4457-8ff2-5e7d1cd86b3d",
            "display": "Westtt Patient",
            "links": [
                {
                    "rel": "self",
                    "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/person/9b6b0edd-5494-43c5-a69d-ddcc5372385b/name/25acf26b-74c6-4457-8ff2-5e7d1cd86b3d",
                    "resourceAlias": "name"
                }
            ]
        },
        "preferredAddress": {
            "uuid": "22803106-5fd1-4367-9d5a-7a3d73b68e73",
            "display": "Chamkar Leu",
            "links": [
                {
                    "rel": "self",
                    "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/person/9b6b0edd-5494-43c5-a69d-ddcc5372385b/address/22803106-5fd1-4367-9d5a-7a3d73b68e73",
                    "resourceAlias": "address"
                }
            ]
        },
        "attributes": [],
        "voided": false,
        "birthtime": null,
        "deathdateEstimated": false,
        "links": [
            {
                "rel": "self",
                "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/person/9b6b0edd-5494-43c5-a69d-ddcc5372385b",
                "resourceAlias": "person"
            },
            {
                "rel": "full",
                "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/person/9b6b0edd-5494-43c5-a69d-ddcc5372385b?v=full",
                "resourceAlias": "person"
            }
        ],
        "resourceVersion": "1.11"
    },
    "voided": false,
    "links": [
        {
            "rel": "self",
            "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/patient/9b6b0edd-5494-43c5-a69d-ddcc5372385b",
            "resourceAlias": "patient"
        },
        {
            "rel": "full",
            "uri": "http://dev3.openmrs.org/openmrs/ws/rest/v1/patient/9b6b0edd-5494-43c5-a69d-ddcc5372385b?v=full",
            "resourceAlias": "patient"
        }
    ],
    "resourceVersion": "1.8"
}
```